src/main/java/Dor.java

```diff
@@ -0,0 +1,10 @@
+public class Dor {
+    public static void main(String[] args) {
+
+        System.out.println("Hello! I'm Dor");
+        System.out.println("What can I do for you?");
+        UserInterface UI = new UserInterface();
+        UI.processInput();
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+}
```

Comment:
Perhaps you can improve this by following the Single Layer of Abstraction Principle (SLAP), as the assignment statement in line 6 is on a different layer of abstraction than the other statements. You can create a method within the same class and place lines 6 and 7 inside. For example : 
```suggestion
public class Dor {
    public static void main(String[] args) {

        System.out.println("Hello! I'm Dor");
        System.out.println("What can I do for you?");
        startListening();
        System.out.println("Bye. Hope to see you again soon!");
    }
    public void startListening() {
        UserInterface UI = new UserInterface();
        UI.processInput();
    }
}
```

Reply:
Implemented, thanks for the suggestion